pkg/controller/namespace/namespace_controller.go

```diff
@@ -155,6 +155,29 @@ func (r *ReconcileNamespace) Reconcile(request reconcile.Request) (reconcile.Res
 			} else {
 				log.WithField("deployment", dep.Name).Info("No suitable Jaeger instances found to inject a sidecar")
 			}
+		} else {
+			// Don't need injection, may be need to remove the sidecar?
+			// If deployment don't have the annotation and has an agent, this may be injected by the namespace
+			// we need to clean it.
+			agent, _ := inject.HasJaegerAgent(dep)
+			_, depAnnotation := dep.Annotations[inject.Annotation]
```

Comment:
`s/depAnnotation/ok` -- you are not getting the annotation, you are getting a boolean on whether it was found.